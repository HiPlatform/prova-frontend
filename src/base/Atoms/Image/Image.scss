img {
  width: auto;
  height: 200px;
}