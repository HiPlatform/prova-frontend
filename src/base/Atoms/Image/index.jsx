import React from 'react';
import PropTypes from 'prop-types';

import './Image.css';

const Image = ({ url, alt }) => (
  <img src={url} alt={alt} />
);

Image.propTypes = {
  url: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
};

export default Image;
