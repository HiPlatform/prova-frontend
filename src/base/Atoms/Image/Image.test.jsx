import React from 'react';
import { shallow } from 'enzyme';
import Image from './index';

describe('Image', () => {
  const props = {
    url: 'http://image.url/given',
    alt: 'some pic',
  };
  const component = shallow(<Image {...props} />);

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('should have the img and alt props rendered', () => {
    expect(component.prop('src')).toBe(props.url);
    expect(component.prop('alt')).toBe(props.alt);
    expect(component.html()).toBe(`<img src="${props.url}" alt="${props.alt}"/>`);
  });
});
