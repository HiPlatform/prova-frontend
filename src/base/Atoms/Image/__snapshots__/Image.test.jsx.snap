// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="some pic"
    url="http://image.url/given"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "some pic",
      "src": "http://image.url/given",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "some pic",
        "src": "http://image.url/given",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
